content/blog/cli-3-1-0/index.Rmd

```diff
@@ -0,0 +1,235 @@
+Our focus will be on improving the cli package in the future, and crayon will only receive important bug fixes.
+If you already use both cli and crayon, then it might make sense to completely switch to cli.
+
+### Palettes in terminals
+
+Many modern terminal emulators, e.g. iTerm on macOS, already allow the customization of ANSI colors, and some also support themes with custom ANSI palettes.
+If you already use this method to customize ANSI colors, then you don't need to set the `cli.palette` option.
+If you use both terminals and RStudio then you can set it only in RStudio:
+
+```{r eval = FALSE}
+if (Sys.getenv("RSTUDIO")=="1") options(cli.palette = "dichro")
+```
+
+## Other improvements
+
+### Bright ANSI colors
+
+cli now has a new set of functions to create the bright version of the 8 base ANSI colors.
+The `col_br_*()` functions set the foreground and the `bg_br_*()` functions set the background colors of strings:
+
+```{r eval = FALSE}
+cli::col_blue("This is blue.")
+cli::col_br_blue("This is bright blue.")
+```
+
+```{asciicast bright, echo = FALSE}
+cli::col_blue("This is blue.")
+cli::col_br_blue("This is bright blue.")
+```
+
+### True color ANSI
+
+cli now supports true color ANSI consoles better.
+Now custom styles made with `make_ansi_style()` will not interpolate the specified color on these displays:
+
+```{r eval = FALSE}
+orange <- make_ansi_style("#eb6123")
+orange("This will be halloween orange.")
+```
+
+```{asciicast orange, echo = FALSE}
+orange <- make_ansi_style("#eb6123")
+orange("This will be halloween orange.")
+```
+
+### Unicode graphemes
+
+cli's `ansi_*()` functions and the new `utf8_*()` functions now handle Unicode graphemes properly.
+For example `ansi_nchar()` and `utf8_nchar()` count graphemes by default, and `ansi_substr()` and `utf8_substr()` will break the input strings at grapheme boundaries.
+
+Consider this Unicode grapheme: 👷🏽‍♀️ (female construction worker, medium skin tone).
+It consists of five Unicode code points:
+
+-   `\U{1f477}`, construction worker,
+-   `\U{1f3fd}`, emoji modifier Fitzpatrick type-4, for the skin tone,
+-   `\u200d`, zero width joiner,
+-   `\u2640`, female sign,
+-   `\ufe0f`, variation selector-16, to specify that the preceding character should be displayed as an emoji.
+
+cli functions handle this grapheme properly:
+
+```{r}
+wrk <- "👷🏽‍♀️"
+as.hexmode(utf8ToInt(wrk))
+```
+
+```{r}
+# graphemes by default
+utf8_nchar(wrk)
+# code points
+utf8_nchar(wrk, type = "codepoints")
+# correct display width
+utf8_nchar(wrk, type = "width")
+```
+
+### Syntax highlight R code
+
+The new `code_highlight()` function parses and syntax highlights R code using ANSI colors and styles.
+You can use `deparse()` to highlight the code of an existing function:
+
+```{r eval = FALSE}
+writeLines(code_highlight(deparse(cli::hash_emoji)))
+```
+
+```{asciicast code, echo = FALSE}
+writeLines(code_highlight(deparse(cli::hash_emoji)))
+```
+
+### Human readable hash functions
```

Comment:
I feel like I'm already seeing these ... in session info, I think? That might make a nice example to illustrate the use case for these.